scripts/filter_detections.py

```diff
@@ -36,133 +74,85 @@
+        detections_gdf = detections_gdf[(detections_gdf['mean_slope'] > MIN_SLOPE_THD) & (detections_gdf['mean_slope'] < MAX_SLOPE_THD)]  
+        total = total - len(detections_gdf)
+        logger.info(f"{total} detections were removed by slope threshold: [{MIN_SLOPE_THD} - {MAX_SLOPE_THD}] degrees")
 
     # Filter dataframe by score value
-    detections = detections[detections['score'] > SCORE]
-    sc = len(detections)
-    logger.info(f"{total - sc} detections were removed by score threshold: {SCORE}")
-
-    # Clip detection to AoI
-    detections = gpd.clip(detections, aoi)
-
-    # Merge close labels using buffer and unions
-    detections_merge = gpd.GeoDataFrame()
-    detections_merge = detections.buffer(+DISTANCE, resolution=2)
-    detections_merge = detections_merge.geometry.unary_union
-    detections_merge = gpd.GeoDataFrame(geometry=[detections_merge], crs=detections.crs)  
-    detections_merge = detections_merge.explode(index_parts=True).reset_index(drop=True)
-    detections_merge = detections_merge.buffer(-DISTANCE, resolution=2)
-
-    td = len(detections_merge)
-    logger.info(f"{td} clustered detections remains after shape union (distance {DISTANCE})")
-
-    # Discard polygons with area under the threshold 
-    detections_merge = detections_merge[detections_merge.area > AREA]
-    ta = len(detections_merge)
-    logger.info(f"{td - ta} detections were removed to after union (distance {AREA})")
-
-    # Preparation of a geo df 
-    data = {'id': detections_merge.index,'area': detections_merge.area, 'centroid_x': detections_merge.centroid.x, 'centroid_y': detections_merge.centroid.y, 'geometry': detections_merge}
-    geo_tmp = gpd.GeoDataFrame(data, crs=detections.crs)
-
-    # Get the averaged detection score of the merged polygons  
-    intersection = gpd.sjoin(geo_tmp, detections, how='inner')
-    intersection['id'] = intersection.index
-    score_final = intersection.groupby(['id']).mean(numeric_only=True)
-
-    # Formatting the final geo df 
-    data = {'id_feature': detections_merge.index,'score': score_final['score'] , 'area': detections_merge.area, 'centroid_x': detections_merge.centroid.x, 'centroid_y': detections_merge.centroid.y, 'geometry': detections_merge}
-    detections_final = gpd.GeoDataFrame(data, crs=detections.crs)
-    logger.info(f"{len(detections_final)} detections remaining after filtering")
+    check_gdf_len(detections_gdf)
+    detections_score_gdf = detections_gdf[detections_gdf.score > SCORE_THD]
+    total = len(detections_gdf) - len(detections_score_gdf)
+    logger.info(f"{total} detections were removed by score filtering (score threshold = {SCORE_THD})")
+
+    # Discard polygons with area under a given threshold 
+    check_gdf_len(detections_gdf)
+    detections_area_gdf = detections_score_gdf.copy()
+    detections_area_gdf['area'] = detections_score_gdf.area
```

Comment:
You already made this attribute at line 98 and I don't think the geometry changed in the meantime.